<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ViewPanel3D</class>
 <widget class="QWidget" name="ViewPanel3D">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>462</width>
    <height>470</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true">* {
font-size: 11px;
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>2</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="GenericView3D" name="view3d" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="toolbar" native="true">
     <property name="styleSheet">
      <string notr="true">QToolButton {
  font-size:11px;
}</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>4</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QToolButton" name="btnUpdateMesh">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>20</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>update</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>10</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>10</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="btnAccept">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>20</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Accept 3D editing operation</string>
        </property>
        <property name="text">
         <string>accept</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnFlip">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>20</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Flip the direction of the cut plane</string>
        </property>
        <property name="text">
         <string>flip</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnCancel">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>20</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Cancel 3D editing operation</string>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>cancel</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="maximumSize">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
        <property name="value">
         <number>24</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="btnColorBar">
        <property name="maximumSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Show or Hide 3D Color Bar</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../Resources/SNAPResources.qrc">
          <normaloff>:/root/color_bar.png</normaloff>:/root/color_bar.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnExpand">
        <property name="maximumSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Expand this view to occupy the entire window</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../Resources/SNAPResources.qrc">
          <normaloff>:/root/dl_3d.png</normaloff>:/root/dl_3d.png</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnScreenshot">
        <property name="maximumSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Save a screenshot</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset>
          <normalon>:/root/icons8-camera-12.png</normalon>
         </iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>12</width>
          <height>12</height>
         </size>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnMenu">
        <property name="maximumSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Additional 3D view controls</string>
        </property>
        <property name="icon">
         <iconset resource="../Resources/SNAPResources.qrc">
          <normaloff>:/root/open_popup_16.png</normaloff>:/root/open_popup_16.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>12</width>
          <height>12</height>
         </size>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
  <action name="actionReset_Viewpoint">
   <property name="text">
    <string>Reset Viewpoint</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, K</string>
   </property>
  </action>
  <action name="actionSave_Viewpoint">
   <property name="text">
    <string>Save Viewpoint</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, S</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::WindowShortcut</enum>
   </property>
  </action>
  <action name="actionRestore_Viewpoint">
   <property name="text">
    <string>Restore Viewpoint</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, R</string>
   </property>
  </action>
  <action name="actionContinuous_Update">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Continuous Update</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, C</string>
   </property>
  </action>
  <action name="actionClear_Rendering">
   <property name="text">
    <string>Clear 3D Display</string>
   </property>
   <property name="toolTip">
    <string>Clears the meshes rendered in the 3D display</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K, X</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GenericView3D</class>
   <extends>QWidget</extends>
   <header>GenericView3D.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../Resources/SNAPResources.qrc"/>
 </resources>
 <connections/>
</ui>
